keras_trainer/trainer.py

```diff
@@ -107,15 +118,41 @@ def __init__(self, **options):
         )
 
         if not self.train_generator:
-            self.train_gen = self.train_data_generator.flow_from_directory(
-                self.train_dataset_dir,
-                batch_size=self.batch_size,
-                target_size=self.model_spec.target_size[:2],
-                class_mode='categorical'
-            )
-            self.num_classes = self.num_classes or self.train_gen.num_classes
-        else:
-            self.train_gen = self.train_generator
+            if self.train_dataset_dataframe is None and os.path.isdir(self.train_dataset_dir):
```

Comment:
Right now, if both `train_dataset_dir` and `train_dataset_dataframe` are provided, we're prioritizing the `train_dataset_dir` value. Is that what we want? Or we should raise an error?

Reply:
`train_dataset_dir` is always required, we are prioritizing `train_dataset_dataframe` if it exists

Reply:
True! Thanks for the simplification 👀